bundles/org.openhab.binding.liquidcheck/src/main/java/org/openhab/binding/liquidcheck/internal/httpclient/LiquidCheckHttpClient.java

```diff
@@ -0,0 +1,107 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.openhab.binding.liquidcheck.internal.LiquidCheckConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LiquidCheckHttpClient} sets up the jetty client for the connection to the device.
+ *
+ * @author Marcel Goerentz - Initial contribution
+ */
+@NonNullByDefault
+public class LiquidCheckHttpClient {
+
+    private final Logger logger = LoggerFactory.getLogger(LiquidCheckHttpClient.class);
+    private final HttpClient client;
+    private final LiquidCheckConfiguration config;
+
+    public boolean isClosed = false;
+
+    /**
+     * The Constructor of the LiquidCheckHttpClient class will set up a jetty client
+     * 
+     * @param config
+     */
+    public LiquidCheckHttpClient(LiquidCheckConfiguration config, HttpClient client) {
+        this.config = config;
+        this.client = client;
+        startClient();
+    }
+
+    /**
+     * The pollData method will poll the data from device
+     * 
+     * @return String with the response of the request
+     * @throws InterruptedException
+     * @throws TimeoutException
+     * @throws ExecutionException
+     */
+    public String pollData() throws InterruptedException, TimeoutException, ExecutionException {
+        String uri = "http://" + config.hostname + "/infos.json";
+        Request request = client.newRequest(uri).method(HttpMethod.GET)
+                .timeout(config.connectionTimeout, TimeUnit.SECONDS).followRedirects(false);
+        ContentResponse response = request.send();
+        return response.getContentAsString();
+    }
+
+    /**
+     * The measureCommand method will start a measurement
+     * 
+     * @return String with response of the request
+     * @throws InterruptedException
+     * @throws TimeoutException
+     * @throws ExecutionException
+     */
+    public String measureCommand() throws InterruptedException, TimeoutException, ExecutionException {
+        String uri = "http://" + config.hostname + "/command";
+        Request request = client.newRequest(uri);
+        request.method(HttpMethod.POST);
+        request.header(HttpHeader.CONTENT_TYPE, "applicaton/json");
+        request.content(new StringContentProvider(
+                "{\"header\":{\"namespace\":\"Device.Control\",\"name\":\"StartMeasure\",\"messageId\":\"1\",\"payloadVersion\":\"1\"},\"payload\":null}"));
+        ContentResponse response = request.send();
+        return response.getContentAsString();
+    }
+
+    public void startClient() {
+        try {
+            client.start();
```

Comment:
The common shared HTTP client is being passed to this class, so there is no need to start it.

Reply:
This is still the case.

Reply:
Deleted